django_lightweight_queue/utils.py

```diff
@@ -47,18 +48,15 @@ def with_prefix(names: Iterable[str]) -> Set[str]:
             for name in names
         )
 
-    setting_names = get_setting_names(app_settings)
+    setting_names = get_setting_names(Defaults())
     extra_names = get_setting_names(extra_settings)
 
     unexpected_names = extra_names - with_prefix(setting_names)
     if unexpected_names:
         unexpected_str = "' ,'".join(unexpected_names)
         warnings.warn("Ignoring unexpected setting(s) '{}'.".format(unexpected_str))
 
-    override_names = extra_names - unexpected_names
-    for name in override_names:
-        short_name = name[len(constants.SETTING_NAME_PREFIX):]
-        setattr(app_settings, short_name, getattr(extra_settings, name))
+    cast(AppSettings, app_settings).add_layer(LongNameAdapter(extra_settings))
```

Comment:
See comment in `app_settings.py` - I reckon we can make this work more cleanly without a `cast`.

Reply:
80d2fd4ab53e1dd60702d8a76fd81bb8c41d8141